src/deluge/model/sample/sample_recorder.h

```diff
@@ -118,11 +116,11 @@ class SampleRecorder {
 
 	int32_t* sourcePos;
 
-	FIL file;
+	std::optional<FatFS::File> file;
 
 private:
 	void setExtraBytesOnPreviousCluster(Cluster* currentCluster, int32_t currentClusterIndex);
-	Error writeCluster(int32_t clusterIndex, int32_t numBytes);
+	Error writeCluster(int32_t clusterIndex, size_t numBytes);
```

Comment:
this is a change from signed to unsigned - is that intentional?

Reply:
yes, this is safe, it's only used when the size is known to be `> 0`